Documented a breaking change regarding imports in @rails/ujs 7.1.0 [ci skip] (#49857)
rails/rails: guides/source/upgrading_ruby_on_rails.md

If you run a single file's tests (`bin/rails test test/models/user_test.rb`), `test:prepare` will not run before it.

### Import syntax from `@rails/ujs` is modified

Starting from Rails 7.1, the syntax for importing modules from `@rails/ujs` is modified. Rails no longer supports the

> direct import of a module from `@rails/ujs`. If users create and build a JavaScript file with the line `import
> { fileInputSelector } from "@rails/ujs"`, they will encounter the error `export 'fileInputSelector'
> (imported as 'fileInputSelector') was not found in '@rails/ujs' (possible exports: default)`.

Reviewer: ```suggestion
direct import of a module from `@rails/ujs`.

For example, attempting to import a function from the library will fail:

\```
import { fileInputSelector } from "@rails/ujs"
#=> export 'fileInputSelector' (imported as 'fileInputSelector') was not found in '@rails/ujs' (possible exports: default)
\```
```

The code fences in the suggested patch might be weird, but the point is I think creating a separate code block for the failure case (with the resulting error on a single line) may help improve readability and anyone searching for this error explicitly.

In Rails 7.1, users should first import the Rails object directly from `@rails/ujs`.
Users can then import specific modules from the Rails object.

An example of imports in Rails 7.1 is shown below:

```javascript
import Rails from "@rails/ujs"
// alias the method
const fileInputSelector = Rails.fileInputSelector

> // reference it from the Rails object where it is used

Reviewer: ```suggestion
// Alternatively, reference it from the Rails object where it is used
```

Author: Thank you very much for the suggestion. I have implemented the suggested change in the updated PR.

If you run a single file's tests (`bin/rails test test/models/user_test.rb`), `test:prepare` will not run before it.

### Import syntax from `@rails/ujs` is modified

Starting from Rails 7.1, the syntax for importing modules from `@rails/ujs` is modified.
Rails no longer supports the direct import
of a module from `@rails/ujs`.
If users create and build a JavaScript file
with the line `import { fileInputSelector } from "@rails/ujs"`,
they will encounter the error `export 'fileInputSelector'
(imported as 'fileInputSelector')
was not found in '@rails/ujs' (possible exports: default)`.

In Rails 7.1, users should first import the Rails object directly from `@rails/ujs`.
Users can then import specific modules

> from the Rails object.

Reviewer: There is some rather peculiar line-wrapping going on here 👀

Author: Good point! I have modified the PR to fix the line-wrapping.

An example of imports in Rails 7.1 is shown below:

```javascript
import Rails from "@rails/ujs"

>  // alias the method

Reviewer 2: ```suggestion
// alias the method
```

const fileInputSelector = Rails.fileInputSelector
// reference it from the Rails object where it is used
Rails.fileInputSelector(...)
```

> 
> 
> 
> 

Reviewer 2: ```suggestion
```

Author: Thank you very much for the suggestions. I have implemented these suggested changes in the updated PR.
